### PR TITLE
Fix IntervalSuite cases failure

### LINKED ISSUE #5497: [BUG] 3 tests in `IntervalSuite` are faling on 330
**Describe the bug**
IntervalSuite:
- test GPU cache interval when reading/writing parquet *** FAILED ***
- test ANSI interval read *** FAILED ***
- test ANSI interval write *** FAILED ***

**Steps/Code to reproduce bug**
`mvn clean test -Dbuildver=330`

### PR DESCRIPTION
**Pull request by res-life**
Fixes #5497

- Update `checkSchemaCompat` for year-month and day-time interval types
- Enable previously failed cases

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala [hunk 1]
```diff
@@ -784,9 +784,10 @@ private case class GpuParquetFileFilterHandler(@transient sqlConf: SQLConf) exte
       case PrimitiveTypeName.BOOLEAN if dt == DataTypes.BooleanType =>
         return
 
-      // TODO: add YearMonthIntervalType
       case PrimitiveTypeName.INT32 =>
-        if (dt == DataTypes.IntegerType || canReadAsIntDecimal(pt, dt)) {
+        if (dt == DataTypes.IntegerType || GpuTypeShims.isSupportedYearMonthType(dt)
```
Comment (sperlingxx) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala:788]: nit: Perhaps it is better to add extra if stmt for these two new types, so as to follow the code style of spark: https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java#L128

**Review by sperlingxx — APPROVED** [review]
LGTM

**Pull request merged**